pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -86,12 +85,123 @@ func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageClus
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResource
+	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	// storageClasses stores all the StorageClasses coming from the JSON
+	var storageClasses map[string]map[string]string = make(map[string]map[string]string)
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace given for ConfigMap?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			cm.Data = d.Data
+			found := &corev1.ConfigMap{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph ConfigMap:	" + cm.Name)
+					err = r.client.Create(context.TODO(), cm)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "Secret":
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace for Secret?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			for k, v := range d.Data {
+				s.Data[k] = []byte(v)
+			}
+			found := &corev1.Secret{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph Secret: 	" + s.Name)
+					err = r.client.Create(context.TODO(), s)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "StorageClass":
+			storageClasses[d.Name] = d.Data
+		}
+	}
+	scs, err := r.newStorageClasses(instance)
+	if err != nil {
+		return err
+	}
+	if len(scs) < 3 {
+		scs = append(scs, r.newOBCStorageClass(instance))
+	}
+	var testStorageClass *storagev1.StorageClass
+	for scName, scData := range storageClasses { //iterating over the map
+		if scName == "cephfs" {
+			// Setting the fsname for cephfs StorageClass
+			testStorageClass = scs[0]
+			// Setting the PoolName cephfs StorageClass
+		} else if scName == "ceph-rbd" {
+			// Setting the PoolName for RBD StorageClass
+			testStorageClass = scs[1]
+		} else if scName == "ceph-rgw" {
+			// Setting the Endpoint for OBC StorageClass
+			testStorageClass = scs[2]
+		}
+		for k, v := range scData {
+			testStorageClass.Parameters[k] = v
+		}
+	}
+	// creating all the storageClass once we set the values
+	err = r.createStorageClasses(scs, reqLogger)
+	if err != nil {
+		return err
+	}
```

Comment:
Don't do *any* of this here, you want all this integrated into `newStorageClasses()`. You need to create a new StorageCluster.Status field and save `storageClasses` there.

Reply:
Here's what I was thinking, something like this.

```
// This if block goes on the very first line of newStorageClasses() 

if StorageCluster.Status.ExterClusterFound {
// Move the Entire code from  line 122 to 200 here in this If block.
}
```

On a side note, what are we going to achieve moving under `newStorageClasses()` here instead of having a separate function?.